backend/Perflow/Services/Implementations/PlaylistService.cs

```diff
@@ -76,5 +89,56 @@ public async Task<int> DeleteEntityAsync(int entityId)
 
             return entityId;
         }
+
+        public async Task<PlaylistSongDTO> AddSongAsync(PlaylistSongDTO playlistSongDTO)
+        {
+            var playlist = await context.Playlists
+                .Where(playlist => playlist.Id == playlistSongDTO.PlaylistId)
+                .FirstOrDefaultAsync();
+
+            var song = await context.Songs
+                .FirstOrDefaultAsync(song => song.Id == playlistSongDTO.SongId);
+
+            var playlistSong = new PlaylistSong {
+                Playlist = playlist,
+                Song = song
+            };
+
+            await context.PlaylistSong.AddAsync(playlistSong);
+
+            await context.SaveChangesAsync();
+
+            return playlistSongDTO;
+        }
+
+        public async Task<PlaylistSongDTO> DeleteSongAsync(PlaylistSongDTO playlistSongDTO)
+        {
+            var playlistSong = await context.PlaylistSong
+                .Where(p => p.PlaylistId == playlistSongDTO.PlaylistId && p.SongId == playlistSongDTO.SongId)
+                .FirstOrDefaultAsync();
```

Comment:
Where + FirstOrDefaultAsync could change to
FirstOrDefaultAsync(p => p.PlaylistId == playlistSongDTO.PlaylistId && p.SongId == playlistSongDTO.SongId)

Reply:
Here and everywhere in your changes